pkg/build/gobuild.go

```diff
@@ -804,6 +823,18 @@ func (g *gobuild) buildOne(ctx context.Context, refStr string, base v1.Image, pl
 		return nil, err
 	}
 
+	annotations := map[string]string{}
+
+	if g.defaultImageConfig != nil {
```

Comment:
I think you could just collapse labels and annotations into the same if/fors:

```
if g.defaultImageConfig != nil {
  for k, v := range labels { ... }
  for k, v := range annotations { ... }
}
if ic := g.imageConfigs[ref.Path()]; ic != {} {
  for k, v := range ic.Labels { ... }
  for k, v := range ic.Annotations { ... }
}
for k, v := range g.labels { ... }
```

Does that make sense?

Reply:
definitely makes sense 🙋🏻‍♂️